### DISCUSSION TITLE
Revalidation of Dynamic Sitemap

### DISCUSSION THREAD

**Discussion by PrinceGupta1999** [vercel/next.js#54974] [Ideas]
### Goals

1. Allow a revalidation logic when sitemaps are generated via [sitemap.ts](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generate-a-sitemap). 
2. The current behaviour is to statically generate the sitemap.xml during build which makes the approach not much different than storing the XML file directly in the public directory

### Non-Goals

1.
2.
3.

### Background

1. Revalidation would allow sitemaps to get automatically refresh if the URLs in the sitemaps are dynamically fetched / generated and new content is added
2. It would also allow for updating the `lastmod` parameter (last modifed time) for existing pages which can help with signaling to crawl bots when an [indexing is needed ](https://developers.google.com/search/docs/crawling-indexing/sitemaps/build-sitemap#additional-notes-about-xml-sitemaps)
3. A current alternative to dynamic sitemaps can probably be to manually generate the XML template for the sitemap using a GET route handler. But that would be an overkill for most applications.

### Proposal

The revalidation logic should be handled similarly to how it is done for any other route

Revalidation
```js
export const revalidate = 14400;
```
Similarly other [specifics of caching and revalidation data](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating) could be followed for sitemap as well.

**Comment by deli-ght** [comment]
Even `force-dynamic` doesn't work neither. I think it is only way to make sitemap.xml/route.ts file for REAL dynamic sitemap
